app/api/leads/[id]/route.ts

```diff
@@ -0,0 +1,82 @@
+import { PrismaClient } from "@prisma/client";
+import apiEndpointWrapper from "utils/api/apiEndpointWrapper";
+import {
+  calculateEstimatedCommission,
+  convertAndRoundToDecimalPlaces,
+  extractLeadId,
+  validateLeadData,
+  validateLeadExistence,
+} from "utils/api/leadUtils";
+
+// LEAD EDIT
+export async function PUT(request: Request) {
+  const reqBody = await request.json();
+
+  return apiEndpointWrapper({
+    method: "PUT",
```

Comment:
See my comment in the API endpoint wrapper for why I created this HOF to wrap all of the endpoints. One thing that jumps out at me even as I look at it now is the redundancy of wrapping the function in PUT to tell NextJS to expose the endpoint but then also passing method:"PUT" to the wrapper.

I did this because the wrapper needs to know what status code to map to on a success response and I wanted to avoid manually adding success status codes for all of the endpoints. But I would have liked to have found a less redundant way to do that.